portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,32 +27,39 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
+import java.util.List;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    ArrayList<String> jsonData = new ArrayList<String>();
+    
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    int numOfComments;
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {   
+        String numOfCommentsString = request.getParameter("numOfComments");
+        if (numOfCommentsString != null) {
+            numOfComments = Integer.parseInt(numOfCommentsString);
```

Comment:
FYI, if `numOfCommentsString` is something like `abc`, then this would raise an exception. APIs usually check to make sure the request params are valid and throw a readable error message if it is not.

In this case, it might be fine, since your input form on the frontend does some validation to make sure it's a number between 1 and 10. I just want to make sure that you think about it when creating these actions and that it's a conscious decision.

Reply:
Should I do something like if (numOfCommentsString >= 1 & numOfCommentsString <= 10) then parse?

Reply:
That wont work because `numOfCommentsString` is a string.

You can do something like
```
try {
  numOfComments  = Integer.parseInt(numOfCommentsString);
} catch(NumberFormatException e){  
  response.setStatus(HttpServletResponse.SC_ERROR);
  response.setContentType("text/html;");
  response.getWriter().println("Error: `numOfComments` is not a string");
} 
# continue with rest of code
```

Reply:
I'm getting an error that it can not find symbol SC_ERROR. Should I try response.sendError(int) instead?